tree/30_generic_methods/package_present.cf

```diff
@@ -0,0 +1,146 @@
+      "action"       string => "warn", if_match_regex => "verify";
+      "action"       string => "immediate", if_match_regex => "";
+      "architecture" string => "default", if_match_regex => "";
+      "version"      string => "any", if_match_regex => "";
+
+  classes:
+      # For now the verify mode does not work, we need a global class
+      "immediate"       expression => strcmp("immediate", "${action}");
+      "warn"            expression => strcmp("warn", "${action}");
+      "version_latest"  expression => strcmp("latest", "${version}");
+      # As "latest" is understood by package promises,
+      # the only special case is "any", which maps to
+      # no specified version in the promise
+      "version_specified"      not => strcmp("any", "${version}");
+      # If architecture is not specified, do not add it to the promise.
+      # The package module will pick the default one for the local platform
+      "architecture_specified" not => strcmp("default", "${architecture}");
+
+  packages:
+
+    debian.architecure_specified.version_specified::
+      "${name}"
+          policy         => "present",
+          package_module => apt_get,
+          architecture   => "${architecture}",
+          version        => "${version}",
+          classes        => classes_generic_two("${old_class_prefix}", "${class_prefix}"),
+          action         => do_if_immediate;
+
+    debian.architecure_specified.!version_specified::
+      "${name}"
+          policy         => "present",
+          package_module => apt_get,
+          architecture   => "${architecture}",
+          classes        => classes_generic_two("${old_class_prefix}", "${class_prefix}"),
+          action         => do_if_immediate;
+
+    debian.!architecure_specified.version_specified::
+      "${name}"
+          policy         => "present",
+          package_module => apt_get,
+          version        => "${version}",
+          classes        => classes_generic_two("${old_class_prefix}", "${class_prefix}"),
+          action         => do_if_immediate;
+
+    debian.!architecure_specified.!version_specified::
+      "${name}"
+          policy         => "present",
+          package_module => apt_get,
+          classes        => classes_generic_two("${old_class_prefix}", "${class_prefix}"),
+          action         => do_if_immediate;
+    
+  methods:
+
+    (debian|redhat)::
+      "reports" usebundle => _log("${action_description} ${name}${architecture_description}${version_description}", "${old_class_prefix}", "${class_prefix}", @{args});
+
+    !(debian|redhat)::
+      "force failure class" usebundle => _classes_failure("${class_prefix}");
+      "force failure class" usebundle => _classes_failure("${old_class_prefix}");
+      "report failure"      usebundle => _log("Package ${name} - systems other than Debian-, and Red Hat-like are not currently supported", "${old_class_prefix}", "${class_prefix}", @{args});
+
+}
+
+body package_module apt_get
```

Comment:
It is still a WIP, I will add Red Hat support after. The body is there for convenience, it will be moved in a "3.7 ncf_lib" after.